core/src/main/java/org/testcontainers/utility/MountableFile.java

```diff
@@ -363,7 +364,15 @@ private int getUnixFileMode(final String pathAsString) {
         }
 
         try {
-            return (int) Files.getAttribute(path, "unix:mode");
+            int unixMode = (int) Files.getAttribute(path, "unix:mode");
+            // Truncate mode bits for z/OS
+            if ("OS/390".equals(SystemUtils.OS_NAME) || 
+                "z/OS".equals(SystemUtils.OS_NAME) || 
+                "zOS".equals(SystemUtils.OS_NAME) ) {
```

Comment:
Just a thought: is it necessary to decide whether to truncate based on system type, or could we always do it?

Linux is our only supported _container OS_, so perhaps the way to think about it could be just as ensuring that no matter what `unixMode` we get, it's always within the right bounds.

Reply:
I thought it would be safest to only truncate for z/OS to minimize potential side effects of this PR. 

I can't think of any reason a Linux distribution would have out of range mode bits, but I think it would be good to capture such a scenario in the form of an error first so we can dissect what's going on, rather than coercing the mode bits into the allowed range and possibly dealing with harder-to-debug side effects as a result.